tests/schema/mysql/test_mysql_schema.py

```diff
@@ -93,6 +93,14 @@ def test_column_exists(self):
         )()
         self.assertEqual(to_sql, sql)
 
+    def test_table_exists(self):
+        to_sql = self.schema.has_table("users", query_only=True)
+
+        sql = getattr(
+            self, inspect.currentframe().f_code.co_name.replace("test_", "")
+        )()
+        self.assertEqual(to_sql, sql)
+
```

Comment:
You'll need a test that looks like this:

```python
    def test_sets_foreign_constraint(self):
        with self.schema.create('users') as table:
             table.string('name')
             table.integer('profile_id')

             table.foreign('profile_id').references('user').on('users')

        sql = getattr(
            self, inspect.currentframe().f_code.co_name.replace("test_", "")
        )()
        self.assertEqual(to_sql, sql)
```

and then make the child test like this:

```python
def sets_foreign_constraint(self):
     return (
        "CREATE TABLE `users` ("
            "`name` VARCHAR(255) NOT NULL, "
            "`profile_id` INT UNSIGNED,  "
                "INDEX profile_users_profile_id, "
                "FOREIGN KEY (`profile_id`)"
                "REFERENCES `profile`(`id`)"
        ")"
    )
```

Reply:
Ok, I will change the tests to assert this.